python-checks/src/main/java/org/sonar/python/checks/UnusedLocalVariableCheck.java

```diff
@@ -208,19 +239,39 @@ private static boolean hasOnlyBindingUsages(Symbol symbol) {
       return false;
     }
     return usages.stream().noneMatch(usage -> usage.kind() == Usage.Kind.IMPORT)
-      && usages.stream().allMatch(Usage::isBindingUsage);
+           && usages.stream().allMatch(Usage::isBindingUsage);
+  }
+
+  private static boolean isNotUpdatingParameterDict(Symbol symbol) {
+    List<Usage> usages = symbol.usages();
+    return usages.stream().noneMatch(UnusedLocalVariableCheck::isDictAssignmentExpressionUsage) ||
+           usages.stream().noneMatch(usage -> usage.kind() == Usage.Kind.PARAMETER);
+  }
+
+  private static boolean isDictAssignmentExpressionUsage(Usage usage) {
+    Tree compoundAssignmentTree = TreeUtils.firstAncestorOfKind(usage.tree(), Kind.COMPOUND_ASSIGNMENT);
+    return compoundAssignmentTree instanceof CompoundAssignmentStatement compoundAssignmentStatement &&
+           "|=".equals(compoundAssignmentStatement.compoundAssignmentToken().value()) &&
+           compoundAssignmentStatement.lhsExpression().type().mustBeOrExtend("dict");
```

Comment:
Here I think we could add a test to cover this case. I know the QG passes already but it should not be too difficult.